CMakeLists.txt

```diff
@@ -5,15 +5,86 @@ project("SVF")
 configure_file(${PROJECT_SOURCE_DIR}/.config.in
                ${PROJECT_BINARY_DIR}/include/Util/config.h)
 
-# We need to match the build environment for LLVM: In particular, we need C++14
-# and the -fno-rtti flag
-set(CMAKE_CXX_STANDARD 14)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
-# add -std=gnu++14
-set(CMAKE_CXX_EXTENSIONS ON)
+# Match configuration used to build LLVM (match C++ standard; match
+# runtime typing information (RTTI); match exception handling; etc)
+if(COMMAND add_llvm_library)
+  message(STATUS "Detected in-tree build configuration; skipping LLVM fetching")
+  set(IN_SOURCE_BUILD 1)
+else()
+  message(STATUS "Detected out-of-tree build configuration; fetching LLVM")
 
-add_compile_options("-fno-rtti")
-add_compile_options("-fno-exceptions")
+  find_package(LLVM REQUIRED CONFIG HINTS ${LLVM_DIR} $ENV{LLVM_DIR})
+  message(STATUS "LLVM STATUS:
+    Version       ${LLVM_VERSION}
+    Definitions   ${LLVM_DEFINITIONS}
+    Includes      ${LLVM_INCLUDE_DIRS}
+    Libraries     ${LLVM_LIBRARY_DIRS}
+    Targets       ${LLVM_TARGETS_TO_BUILD}
+    Build type    ${LLVM_BUILD_TYPE}
+    Exceptions    ${LLVM_ENABLE_EH}
+    RTTI          ${LLVM_ENABLE_RTTI}
+    Dynamic lib   ${LLVM_LINK_LLVM_DYLIB}"
+  )
```

Comment:
After a second look, the current cmakelist in the root folder was not supposed to have anything related to LLVM and those LLVM options should be moved to svf-llvm/CMakelist.txt. I assume this change should be straight-forward?

SVF itself is now designed language-independent, and we may have another module later (e.g., svf-xxx) which may not rely on llvm-based language, so a clean root cmakelist would be helpful.

Reply:
Alright; I moved this check to the top-level `CMakeLists.txt` because that was where flags like `-fno-exceptions` or `-fno-rtti` were added. I've moved this check back to the script in the `svf-llvm` directory; the flags for `-fno-rtti` or `-fno-exceptions` are now only added to `SvfLLVM` (if users want the main library to be built with it they can just specify it through `$CFLAGS` or `$CXXFLAGS`).